docs/adr/007_cups_firmware_upgrade.md

```diff
@@ -0,0 +1,211 @@
+# 007. CUPS Protocol Implementation - Firmware Upgrade
+
+**Feature**:
+[#1189](https://github.com/Azure/iotedge-lorawan-starterkit/issues/1189)  
+
+**Authors**: Daniele Antonio Maggio, Maggie Salak
+
+**Status**: Proposed
+
+>This ADR is an extension of [006. CUPS Protocol Implementation - Credential
+>management][cups-adr] and focuses of firmware upgrades. For details about the
+>general CUPS protocol implementation please refer to the other document.
+
+## Overview
+
+Firmware upgrades for LoRa Basics™ Station need to be supported in the CUPS
+protocol implementation. More information on the protocol and parameters
+exchanged between the Station and the CUPS server which are relevant to firmware
+upgrades can be found in [The CUPS protocol - documentation][cupsproto].
+
+## In-scope
+
+This document focuses on:
+
+- Defining a flow for executing firmware upgrades of the Basics Station
+- Defining the changes needed in device twins for the concentrator station
+  stored in IoT Hub for handling firmware upgrades
+- Defining the changes required in the storage solution used for the CUPS
+  protocol implementation
+- Defining the changes needed in the Azure Function for supporting firmware
+  upgrades
+- Defining the changes needed in LoRaWan Network Server (LNS) for handling
+  firmware upgrades
+- Defining the changes needed in LoRa Device Provisioning CLI for allowing
+  firmware upgrades
+
+## Out-of-scope
+
+- Generating of the signature key, CRC32 checksum of the signature and digest in
+  the LoRa Device Provisioning CLI is considered a stretch and will not be added
+  to the tool as a functionality for the time being. This document provides
+  sample commands which can be used to generate required values in the appendix.
+
+## Decisions
+
+### Context
+
+The CUPS request described in [The CUPS protocol - documentation][cupsproto]
+contains the `package` field which indicates the current firmware version of the
+Basics Station. The value will need to be updated whenever a firmware upgrade of
+the Basics Station is performed by the user.
+
+### Firmware upgrade flow
+
+1. When provisioning a device that will need firmware upgrades, the end user
+   should generate a `sig-0.key` and store it on the device and in a centralized
+   repository of their own choice (example on how to generate it can be found in
+   the appendix).
+
+1. When a firmware upgrade needs to be provided to the device, the user should
+   generate a digest of the executable file of the upgrade (as in examples
```

Comment:
nit: is it an executable? or just a binary?

Reply:
it must be an executable, as per cups protocol specification.